src/Generators/randomgenerators.jl

```diff
@@ -0,0 +1,407 @@
+abstract type RandomGenerator <: GraphGenerator end
+
+"""
+    struct ErdosRenyi <: RandomGenerator
+
+A struct representing a generator for an [Erdős–Rényi](http://en.wikipedia.org/wiki/Erdős–Rényi_model)
+random graph with `n` vertices. Edges are added between pairs of vertices with
+probability `p`.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the graph
+- `p::Real`: the probability of edge creation (`0 < p ≤ 1`)
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+"""
+struct ErdosRenyi{T<:Integer, U<:Real, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    p::U
+    rng::R
+end
+ErdosRenyi(n, p; rng=GLOBAL_RNG) = ErdosRenyi(n, p, rng)
+
+"""
+    struct ApproxErdosRenyi <: RandomGenerator
+
+Create a generator approximating an [Erdős–Rényi](http://en.wikipedia.org/wiki/Erdős–Rényi_model) random
+graph with `n` vertices and `ne` edges.
+
+### Required Fields
+- `n::Integer`: the number of vertices in the graph
+- `ne::Integer`: the number of edges in the graph
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+"""
+struct ApproxErdosRenyi{T<:Integer, U<:Integer, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    ne::U
+    rng::R
+end
+ApproxErdosRenyi(n, ne; rng=GLOBAL_RNG) = ApproxErdosRenyi(n, ne, rng)
+
+"""
+    struct ExpectedDegree <: RandomGenerator 
+
+A struct representing a generator for a random undirected graph in which two vertices `i` and `j` are connected
+with a probability (given by a vector of expected degrees `ω` indexed by vertex) of `ω[i]*ω[j]/sum(ω)`.
+
+### Required Fields
+- `ω::AbstractVector{<:Real}`: the probability vector for vertex connection
+
+### Optional Arguments
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+
+### Implementation Notes
+The algorithm should work well for `maximum(ω) ≪ sum(ω)`. As `maximum(ω)` approaches `sum(ω)`, some deviations
+from the expected values are likely.
+
+### References
+- Connected Components in Random Graphs with Given Expected Degree Sequences, Linyuan Lu and Fan Chung. [https://link.springer.com/article/10.1007%2FPL00012580](https://link.springer.com/article/10.1007%2FPL00012580)
+- Efficient Generation of Networks with Given Expected Degrees, Joel C. Miller and Aric Hagberg. [https://doi.org/10.1007/978-3-642-21286-4_10](https://doi.org/10.1007/978-3-642-21286-4_10)
+"""
+struct ExpectedDegree{T<:Real, U<:AbstractVector{T}, R<:AbstractRNG} <: RandomGenerator
```

Comment:
Can we just use 
```julia
    ExpectedDegrees{U <: AbstractVector{<: Real}, R < AbstractRNG}
```
?

Reply:
Yes, but I wanted to be consistent in signature for all the generators. Some implementation might need`T` at some point.